test/core/security/local_security_connector_test.cc

```diff
@@ -0,0 +1,165 @@
+                                       grpc_pollset_set* /*pollset*/) {}
+
+static void me_destroy(grpc_endpoint* ep) {}
+
+static absl::string_view me_get_peer(grpc_endpoint* /*ep*/) {
+  return "";
+}
+
+static absl::string_view me_get_local_address_unix(grpc_endpoint* /*ep*/) {
+  return "unix:";
+}
+
+static int me_get_fd(grpc_endpoint* /*ep*/) { return -1; }
+
+static bool me_can_track_err(grpc_endpoint* /*ep*/) { return false; }
+
+static const grpc_endpoint_vtable vtable_unix = {me_read,
+                                            me_write,
+                                            me_add_to_pollset,
+                                            me_add_to_pollset_set,
+                                            me_delete_from_pollset_set,
+                                            me_destroy,
+                                            me_get_peer,
+                                            me_get_local_address_unix,
+                                            me_get_fd,
+                                            me_can_track_err};
+
+static absl::string_view me_get_local_address_local(grpc_endpoint* /*ep*/) {
+  return "ipv4:127.0.0.1:12667";
+}
+
+static const grpc_endpoint_vtable vtable_local = {me_read,
+                                            me_write,
+                                            me_add_to_pollset,
+                                            me_add_to_pollset_set,
+                                            me_delete_from_pollset_set,
+                                            me_destroy,
+                                            me_get_peer,
+                                            me_get_local_address_local,
+                                            me_get_fd,
+                                            me_can_track_err};
+
+static void check_tsi_security_level(grpc_local_connect_type connect_type,
+                                 tsi_security_level level, grpc_endpoint ep) {
+  grpc_server_credentials* server_creds = grpc_local_server_credentials_create(connect_type);
+  ChannelArgs args;
+  EXPECT_NE(server_creds, nullptr);
+  RefCountedPtr<grpc_server_security_connector> connector = server_creds->
+      create_security_connector(args);
+  EXPECT_NE(connector, nullptr);
+  tsi_peer peer;
+  CHECK(tsi_construct_peer(0, &peer) == TSI_OK);
+
+  RefCountedPtr<grpc_auth_context> auth_context;
+  connector->check_peer(peer, &ep, args, &auth_context, nullptr);
+  tsi_peer_destruct(&peer);
+  auto it = grpc_auth_context_find_properties_by_name(
+      auth_context.get(), GRPC_TRANSPORT_SECURITY_LEVEL_PROPERTY_NAME);
+  const grpc_auth_property* prop = grpc_auth_property_iterator_next(&it);
+  ASSERT_NE(prop, nullptr);
+  EXPECT_STREQ(prop->value, tsi_security_level_to_string(level));
+  connector.reset();
+  auth_context.reset();
+  grpc_server_credentials_release(server_creds);
+}
+
+//
+// Tests for Certificate Providers in ChannelSecurityConnector.
```

Comment:
Please remove this comment, as it is not relevant here.

Reply:
Changed the comment to `Tests for grpc_local_channel_security_connector.` - in surrounding tests it's used as a separator between test util code and tests themselves. Is it better to remove it completely?

Reply:
Thanks for fixing the wording. I generally omit this comment, and only include it if there are a huge number of very different tests in the same test file. I'll leave it to you whether you want to include it.